SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/WorkerCoordinator/LifetimeComponent.cs

```diff
@@ -0,0 +1,208 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+using System;
+using System.Threading;
+using System.Collections.Generic;
+
+namespace Improbable.WorkerCoordinator
+{
+    /// <summary>
+    /// Simulated player client's information.
+    /// The coordinator use this to start & restart simulated player clients.
+    /// </summary>
+    struct ClientInfo
+    {
+        public string ClientName;
+        public string DevAuthToken;
+        public string TargetDeployment;
+        public long StartTick;
+        public long EndTick;
+    }
```

Comment:
Can we maybe do this and use `SimulatedPlayerInstance.Kill()` to stop the simplayer process? This way we won't need to add new scripts.

```suggestion
    struct ClientInfo
    {
        public string ClientName;
        public string DevAuthToken;
        public string TargetDeployment;
        public long StartTick;
        public long EndTick;
        public Process SimulatedPlayerInstance;
    }
```

Reply:
We use `StartSimulatedPlayer.sh` to start the simulated clients. The call sequence is `StartSimulatedPlayer.sh` -> `GDKTestGyms.sh` -> `GDKTestGyms`. We can not get the real process easily.
We use shell script to start clients because the NFR environment have a step to modify it.
That is why I use `StopSimulatedClient.sh` to kill the clients with `ClientName`.

Reply:
We do have a Process that we have started for the `StartSimulatedPlayer.sh`, don't we? Can you check if calling Process.Kill on it achieves the same result?

Reply:
My first version is `Process.Kill()` and was blocked by kill parent but the child still alive. We use `.Net Framework` which does not have `Process.Kill(true);  // .Net core has this` to kill all child/descendent processes. But we still can implement our own method like `KillProcessAndChildren`. I have tried this and it works fine.

The real reason I choose `StopSimulatePlayer.sh` is the `NFR` environment has some step to modify `StartSimulatedPlayer.sh`[(code link)](https://github.com/improbable/TestGymBuildkite/blob/889bd694cde2475d679f358ae5619cead8bdf3da/ci/modify-and-upload-assembly.ps1#L68-L72). That means it is hard to change it and use C# code to replace `StartSimulatedPlayer.sh`. I think it is better to keep this `Start & Stop`  logic in the same way. It is easier to manage. Since we can not `Start` in C#, we can not `Stop` in the same way.